spring-context/src/main/java/org/springframework/context/annotation/ComponentScan.java

```diff
@@ -86,6 +90,13 @@
 	 * attribute.
 	 * <p>Use {@link #basePackageClasses} for a type-safe alternative to
 	 * String-based package names.
+	 * <p>Supports {@code ${...}} placeholders resolved against the
+	 * {@link org.springframework.core.env.Environment Environment} as well as
+	 * Ant-style package patterns (for example, {@code com.example.**}).
+	 * Multiple packages and/or patterns may be specified.
+	 * <p><strong>Note:</strong> Ant-style patterns are <em>not</em> applicable to
+	 * {@link #basePackageClasses()}, which accepts concrete classes for type-safe
+	 * package selection.
```

Comment:
I don't think there is any need to point this out, since one can only supply `Class` references via the `basePackageClasses` attribute.

Thus, I have removed this and other such text locally.